mongo/integration/unified/bulkwrite_helpers.go

```diff
@@ -176,7 +183,7 @@ func createBulkWriteModel(rawModel bson.Raw) (mongo.WriteModel, error) {
 			return nil, newMissingArgumentError("filter")
 		}
 
-		return mongo.NewDeleteOneModel().SetFilter(filter), nil
+		return mongo.NewDeleteOneModel().SetFilter(filter).SetHint(hint), nil
```

Comment:
I don't know if we want to set hint if it isn't included. We could do something like `deleteMany` instead where the model gets initialized before we iterate over the elements.

Reply:
Good point! Copied the logic from `deleteMany` and created the model before iterating over potential options.